From ac50eb66eef7ad8b32e36baa0ad276c0ea77ca10 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 31 Jul 2015 15:22:16 +0000
Subject: [PATCH 0597/1000] MANTIS 36789 Mantis 36789: Platform sending bagtag
 comm error rather than out of stock In ETX-ACk protocol DLE not just relevant
 characters, but all between 0x00 and 0x20

git-svn-id: svn://localhost/SelfServiceCommon/trunk@877 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Interfaces/inc/serialCon/mPEtxAck.hpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SelfServiceCommon/Interfaces/inc/serialCon/mPEtxAck.hpp b/SelfServiceCommon/Interfaces/inc/serialCon/mPEtxAck.hpp
index 11948344..acac6537 100644
--- a/SelfServiceCommon/Interfaces/inc/serialCon/mPEtxAck.hpp
+++ b/SelfServiceCommon/Interfaces/inc/serialCon/mPEtxAck.hpp
@@ -139,6 +139,7 @@ private:
          {
             char C = *Iter;
             TxLRC ^= C;
+#if 0
             switch( C )
             {
             case ASCII_STX:
@@ -152,6 +153,12 @@ private:
             default:
                break;
             }
+#else
+            if( ( C >= 0 ) && ( C < ' ' ) )
+            {
+               M_CharInOut.write( ASCII_DLE );
+            }
+#endif
             M_CharInOut.write( static_cast<unsigned char>( C ) );
             ++Iter;
          }
-- 
2.41.0.windows.1

